drivers/opencl/src/main/java/uk/ac/manchester/tornado/drivers/opencl/graal/compiler/OCLCompilationResultBuilder.java

```diff
@@ -179,6 +179,30 @@ public Value doValue(LIRInstruction instruction, Value value, OperandMode mode,
 
     }
 
+    /**
+     * Checks if there are any LIR instructions between the loop condition and the {@link LoopInitOp} and {@link LoopPostOp}.
+     * If there are no instructions, it is possible to move the loop condition to the loop header.
+     * @return true if there are no instructions.
+     */
+    private static boolean shouldFormatLoopHeader(List<LIRInstruction> instructions) {
+        int loopInitOpIndex = -1, loopPostOpIndex = -1, loopConditionOpIndex = -1;
+
+        for (int index = 0, instructionsSize = instructions.size(); index < instructionsSize; index++) {
+            LIRInstruction instruction = instructions.get(index);
+            if (instruction instanceof LoopInitOp) {
+                loopInitOpIndex = index;
+            }
+            if (instruction instanceof LoopPostOp) {
+                loopPostOpIndex = index;
+            }
+            if (instruction instanceof LoopConditionOp) {
+                loopConditionOpIndex = index;
+            }
+        }
+
+        return (loopConditionOpIndex - 1 == loopPostOpIndex) && (loopConditionOpIndex - 2 == loopInitOpIndex);
```

Comment:
would it make sense to use two constants for the subtraction value? Can you explain why `loopConditionOpIndex` is subtracted by 1, while `loopConditionOpIndex` is by 2?

Reply:
I could use a constant but I can't come up with a suitable name. 

If there are no nodes (fixed or floating) between `LoopBegin` and the `If` corresponding to the loop condition, then in the LIR, the `LoopInitOp` and the `LoopPostOp` will be immediately followed by a `LoopConditionOp`

That is, the `emitLoopBegin` will be called right before the `emitIf` is called.
https://github.com/beehive-lab/TornadoVM/blob/202a27b0ee1f58368feab95619d9e108e7b52cda/drivers/opencl/src/main/java/uk/ac/manchester/tornado/drivers/opencl/graal/compiler/OCLNodeLIRBuilder.java#L505
https://github.com/beehive-lab/TornadoVM/blob/202a27b0ee1f58368feab95619d9e108e7b52cda/drivers/opencl/src/main/java/uk/ac/manchester/tornado/drivers/opencl/graal/compiler/OCLNodeLIRBuilder.java#L471

Reply:
Thanks @gigiblender, lets move the condition to a new method and add a Javadoc description.